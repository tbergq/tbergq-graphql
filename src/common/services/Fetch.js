// @flow

import fetch from 'node-fetch';

export default async (url: string, options: Object = {}) => {
  try {
    const response = await fetch(url, options);

    const json = await response.json();
    return json;
  } catch (err) {
    console.log(err);
    throw err;
  }
};
