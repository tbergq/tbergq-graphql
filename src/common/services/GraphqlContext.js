// @flow

import DataLoader from 'dataloader';

import Userloader from '../../dataloader/Userloader';
import type { UserType } from '../db/UserModel';
import SearchTvShowLoader, {
  type TvShow,
} from '../../tvhelper/dataloaders/SearchTvShowLoader';
import TvDetailLoader from '../../tvhelper/dataloaders/TvDetailLoader';

export type GraphqlContextType = {|
  +dataLoader: {|
    +user: DataLoader<string, UserType>,
    +tvhelper: {|
      +searchTvShow: DataLoader<string, TvShow[]>,
      +tvDetail: DataLoader<string, TvShow>,
    |},
  |},
|};

export default function createContext() {
  return {
    dataLoader: {
      user: Userloader,
      tvhelper: {
        searchTvShow: SearchTvShowLoader(),
        tvDetail: TvDetailLoader(),
      },
    },
  };
}
