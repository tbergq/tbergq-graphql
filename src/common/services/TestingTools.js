// @flow

import { graphql as originalGraphQL } from 'graphql';
import schema from '../../Schema';
import createContext from './GraphqlContext';

export const graphql = async (
  query: string,
  variables: ?Object,
): Promise<Object> =>
  originalGraphQL(schema, query, null, createContext(), variables);
