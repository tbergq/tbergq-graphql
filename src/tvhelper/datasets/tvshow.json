{
    "id": 139,
    "url": "http://www.tvmaze.com/shows/139/girls",
    "name": "Girls",
    "type": "Scripted",
    "language": "English",
    "genres": [
        "Drama",
        "Romance"
    ],
    "status": "Ended",
    "runtime": 30,
    "premiered": "2012-04-15",
    "officialSite": "http://www.hbo.com/girls",
    "schedule": {
        "time": "22:00",
        "days": [
            "Sunday"
        ]
    },
    "rating": {
        "average": 6.7
    },
    "weight": 94,
    "network": {
        "id": 8,
        "name": "HBO",
        "country": {
            "name": "United States",
            "code": "US",
            "timezone": "America/New_York"
        }
    },
    "webChannel": null,
    "externals": {
        "tvrage": 30124,
        "thetvdb": 220411,
        "imdb": "tt1723816"
    },
    "image": {
        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/31/78286.jpg",
        "original": "http://static.tvmaze.com/uploads/images/original_untouched/31/78286.jpg"
    },
    "summary": "<p>This Emmy winning series is a comic look at the assorted humiliations and rare triumphs of a group of girls in their 20s.</p>",
    "updated": 1543140952,
    "_links": {
        "self": {
            "href": "http://api.tvmaze.com/shows/139"
        },
        "previousepisode": {
            "href": "http://api.tvmaze.com/episodes/1079686"
        }
    },
    "_embedded": {
        "cast": [
            {
                "person": {
                    "id": 27410,
                    "url": "http://www.tvmaze.com/people/27410/lena-dunham",
                    "name": "Lena Dunham",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1986-05-13",
                    "deathday": null,
                    "gender": "Female",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/3/7597.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/3/7597.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/27410"
                        }
                    }
                },
                "character": {
                    "id": 36886,
                    "url": "http://www.tvmaze.com/characters/36886/girls-hannah-horvath",
                    "name": "Hannah Horvath",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1954.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1954.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36886"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 11102,
                    "url": "http://www.tvmaze.com/people/11102/allison-williams",
                    "name": "Allison Williams",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1988-04-13",
                    "deathday": null,
                    "gender": "Female",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/3/7628.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/3/7628.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/11102"
                        }
                    }
                },
                "character": {
                    "id": 36887,
                    "url": "http://www.tvmaze.com/characters/36887/girls-marnie-michaels",
                    "name": "Marnie Michaels",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1955.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1955.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36887"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 27411,
                    "url": "http://www.tvmaze.com/people/27411/jemima-kirke",
                    "name": "Jemima Kirke",
                    "country": {
                        "name": "United Kingdom",
                        "code": "GB",
                        "timezone": "Europe/London"
                    },
                    "birthday": "1985-04-26",
                    "deathday": null,
                    "gender": "Female",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/3/7631.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/3/7631.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/27411"
                        }
                    }
                },
                "character": {
                    "id": 36888,
                    "url": "http://www.tvmaze.com/characters/36888/girls-jessa-johansson",
                    "name": "Jessa Johansson",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1957.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1957.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36888"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 16405,
                    "url": "http://www.tvmaze.com/people/16405/zosia-mamet",
                    "name": "Zosia Mamet",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1988-02-02",
                    "deathday": null,
                    "gender": "Female",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/3/7634.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/3/7634.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/16405"
                        }
                    }
                },
                "character": {
                    "id": 36889,
                    "url": "http://www.tvmaze.com/characters/36889/girls-shoshanna-shapiro",
                    "name": "Shoshanna Shapiro",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1956.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1956.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36889"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 24858,
                    "url": "http://www.tvmaze.com/people/24858/adam-driver",
                    "name": "Adam Driver",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1983-11-19",
                    "deathday": null,
                    "gender": "Male",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/170/425572.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/170/425572.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/24858"
                        }
                    }
                },
                "character": {
                    "id": 36890,
                    "url": "http://www.tvmaze.com/characters/36890/girls-adam-sackler",
                    "name": "Adam Sackler",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1959.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1959.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36890"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 24970,
                    "url": "http://www.tvmaze.com/people/24970/alex-karpovsky",
                    "name": "Alex Karpovsky",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1975-09-23",
                    "deathday": null,
                    "gender": "Male",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/3/7638.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/3/7638.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/24970"
                        }
                    }
                },
                "character": {
                    "id": 36891,
                    "url": "http://www.tvmaze.com/characters/36891/girls-ray-ploshansky",
                    "name": "Ray Ploshansky",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1958.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1958.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36891"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 27412,
                    "url": "http://www.tvmaze.com/people/27412/andrew-rannells",
                    "name": "Andrew Rannells",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1978-08-23",
                    "deathday": null,
                    "gender": "Male",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/0/1960.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/0/1960.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/27412"
                        }
                    }
                },
                "character": {
                    "id": 36892,
                    "url": "http://www.tvmaze.com/characters/36892/girls-elijah-krantz",
                    "name": "Elijah Krantz",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/15/38615.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/15/38615.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/36892"
                        }
                    }
                },
                "self": false,
                "voice": false
            },
            {
                "person": {
                    "id": 4896,
                    "url": "http://www.tvmaze.com/people/4896/ebon-moss-bachrach",
                    "name": "Ebon Moss-Bachrach",
                    "country": {
                        "name": "United States",
                        "code": "US",
                        "timezone": "America/New_York"
                    },
                    "birthday": "1977-03-19",
                    "deathday": null,
                    "gender": "Male",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/147/369955.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/147/369955.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/people/4896"
                        }
                    }
                },
                "character": {
                    "id": 37030,
                    "url": "http://www.tvmaze.com/characters/37030/girls-desi",
                    "name": "Desi",
                    "image": {
                        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/15/38616.jpg",
                        "original": "http://static.tvmaze.com/uploads/images/original_untouched/15/38616.jpg"
                    },
                    "_links": {
                        "self": {
                            "href": "http://api.tvmaze.com/characters/37030"
                        }
                    }
                },
                "self": false,
                "voice": false
            }
        ]
    }
}