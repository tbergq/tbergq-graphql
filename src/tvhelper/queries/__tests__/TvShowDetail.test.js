// @flow

import { toGlobalId } from 'graphql-relay';

import { graphql } from '../../../common/services/TestingTools';
import tvshow from '../../datasets/tvshow.json';

it('works', async () => {
  // $FlowFixMe
  fetch.mockResponse(JSON.stringify(tvshow));
  const query = `
  query detail($id: ID!) {
    tvShowDetail(id: $id) {
      id
      name
      status
      premiered
      rating
      image {
        original
        medium
      }
      summary
    }
  }
  `;
  expect(await graphql(query, { id: toGlobalId('test', '123') })).toMatchSnapshot();
});
