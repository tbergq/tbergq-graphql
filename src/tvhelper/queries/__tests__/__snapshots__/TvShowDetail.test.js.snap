// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "data": Object {
    "tvShowDetail": Object {
      "id": "dHZTaG93OjEzOQ==",
      "image": Object {
        "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/31/78286.jpg",
        "original": "http://static.tvmaze.com/uploads/images/original_untouched/31/78286.jpg",
      },
      "name": "Girls",
      "premiered": "2012-04-15",
      "rating": 6.7,
      "status": "Ended",
      "summary": "This Emmy winning series is a comic look at the assorted humiliations and rare triumphs of a group of girls in their 20s.",
    },
  },
}
`;
